src/pages/memos/[functionId]/MemosForm.js

```diff
@@ -0,0 +1,531 @@
+                    error={formik.touched.vatPct && Boolean(formik.errors.vatPct)}
+                  />
+                </Grid>
+                <Grid item xs={12}>
+                  <CustomNumberField
+                    name='vatAmount'
+                    readOnly
+                    label={labels.vatAmount}
+                    value={formik.values.vatAmount}
+                    maxAccess={maxAccess}
+                    onChange={formik.handleChange}
+                    onClear={() => formik.setFieldValue('vatAmount', '')}
+                    error={formik.touched.vatAmount && Boolean(formik.errors.vatAmount)}
+                  />
+                </Grid>
+                <Grid item xs={12}>
+                  <FormControlLabel
+                    control={
+                      <Checkbox
+                        name='isSubjectToVAT'
+                        readOnly={postedOrCanceled}
+                        maxAccess={maxAccess}
+                        checked={formik.values?.isSubjectToVAT}
+                        onChange={!postedOrCanceled ? formik.handleChange : ''}
+                      />
+                    }
+                    label={labels.vat}
+                  />
+                </Grid>
+                <Grid item xs={12}>
+                  <CustomNumberField
+                    name='amount'
+                    type='text'
+                    readOnly
+                    label={labels.amount}
+                    value={formik.values.amount}
+                    required
+                    maxAccess={maxAccess}
+                    onChange={e => formik.setFieldValue('amount', e.target.value)}
+                    onClear={() => formik.setFieldValue('amount', '')}
+                    error={formik.touched.amount && Boolean(formik.errors.amount)}
+                    maxLength={10}
+                  />
+                </Grid>
+              </Grid>
+            </Grid>
+            <Grid item xs={12}>
+              <ResourceLookup
+                endpointId={FinancialRepository.Account.snapshot}
+                parameters={{
+                  _type: 2
+                }}
+                readOnly={postedOrCanceled}
+                valueField='reference'
+                displayField='name'
+                name='accountId'
+                required
+                label={labels.account}
+                form={formik}
+                valueShow='accountRef'
+                secondValueShow='accountName'
+                columnsInDropDown={[
+                  { key: 'reference', value: 'Reference' },
+                  { key: 'name', value: 'Name' },
+                  { key: 'keywords', value: 'Keywords' },
+                  { key: 'groupName', value: 'Account Group' }
+                ]}
+                onChange={(event, newValue) => {
+                  if (newValue) {
+                    formik.setFieldValue('accountId', newValue?.recordId || '')
```

Comment:
remove if else blocks and use conditional ternary operator for less code please
example : formik.setFieldValue('accountRef', newValue? newValue.reference : '')